src/Kiota.Builder/Extensions/OpenApiSchemaExtensions.cs

```diff
@@ -47,10 +47,19 @@ public static string GetSchemaName(this OpenApiSchema schema)
         return schema.GetSchemaNames().LastOrDefault()?.TrimStart('$') ?? string.Empty;// OData $ref
     }
 
+    public static OpenApiSchema? GetSchema(this OpenApiSchema schema)
+    {
+        if (schema is null)
+            return null;
+
+        return schema.IsArray() ? schema.Items : schema;
+    }
+
     public static bool IsReferencedSchema(this OpenApiSchema schema)
     {
-        var isReference = schema?.Reference != null;
-        if (isReference && schema!.Reference.IsExternal)
+        var currentSchema = schema.GetSchema();
+        var isReference = currentSchema?.Reference != null;
+        if (isReference && currentSchema!.Reference.IsExternal)
```

Comment:
this implementation is meant to look at the current schema only and not looking downwards.
I worry that change might derail some logic for edge cases down the road. Please revert it.